modules/tts/epub3-tts/src/main/resources/xml/prepare-for-tts.xsl

```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<xsl:stylesheet
+    version="1.0"
+    xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+    xmlns:html="http://www.w3.org/1999/xhtml"
+    xmlns="http://www.w3.org/1999/xhtml"
+    exclude-result-prefixes="#all">
+
+    <xsl:output method="html" indent="yes"/>
+    <xsl:strip-space elements="html:li"/>
+
+    <xsl:template match="@*|node()">
+        <xsl:copy><xsl:apply-templates select="@*|node()"/></xsl:copy>
+    </xsl:template>
+
+    <xsl:template match="html:ol[not(@class='plain') and not(ancestor::html:ol[@class='plain'])]">
+        <xsl:variable name="startpos">
+            <xsl:choose>
+                <xsl:when test="@start"><xsl:value-of select="@start - 1"/></xsl:when>
+                <xsl:otherwise>0</xsl:otherwise>
+            </xsl:choose>
+        </xsl:variable>
+        <xsl:variable name="format_type">
+            <xsl:choose>
+                <xsl:when test="@type"><xsl:value-of select="@type"/></xsl:when>
+                <xsl:otherwise>1</xsl:otherwise>
+            </xsl:choose>
+        </xsl:variable>
+        <ol>
+            <xsl:copy-of select="@*" />
+            <xsl:for-each select="./html:li">
+                <xsl:variable name="number" select="$startpos + position()"/>
+                <xsl:variable name="number_format">
+                    <xsl:choose>
+                        <xsl:when test="$format_type = 'a'"><xsl:number format="a" value="$number"/></xsl:when>
+                        <xsl:when test="$format_type = 'A'"><xsl:number format="A" value="$number"/></xsl:when>
+                        <xsl:when test="$format_type = 'i'"><xsl:number format="i" value="$number"/></xsl:when>
+                        <xsl:when test="$format_type = 'I'"><xsl:number format="I" value="$number"/></xsl:when>
+                        <xsl:otherwise><xsl:number format="1" value="$number"/></xsl:otherwise>
+                    </xsl:choose>
+                </xsl:variable>
+                <li><xsl:copy-of select="@*" /><span class="formatted_number"><xsl:value-of select="concat($number_format, '.', ' ')"/></span>
```

Comment:
Just use the `tts:speech-only` attribute (with is handled [here](https://github.com/daisy/pipeline/blob/ba10a7abf3612bc4df55b547ec3df7330def5b72/modules/tts/epub3-tts/src/main/resources/xml/tts-for-epub3.xpl#L283-L295)). Then you can be sure that nothing gets removed that shouldn't.

Reply:
Hi @bertfrees 

Great suggestion, worked fine. Have changed to use that attribute instead.

Best regards
Daniel